src/System.Management.Automation/commands/utility/FormatAndOutput/out-console/OutConsole.cs

```diff
@@ -98,7 +96,7 @@ protected override void ProcessRecord()
         {
             if (Transcript)
             {
-                Host.UI.TranscribeOnly = true;
+                Host.UI.TranscribeOnlyCount++;
```

Comment:
You are doing this in process block, once for each input item, not once per `Out-Default` invocation.